airflow/jobs/scheduler_job_runner.py

```diff
@@ -717,11 +717,15 @@ def _critical_section_enqueue_task_instances(self, session: Session) -> int:
         # across all executors.
         num_occupied_slots = sum([executor.slots_occupied for executor in self.job.executors])
         parallelism = conf.getint("core", "parallelism")
+        # Parallelism configured to 0 means infinite currently running tasks
+        if parallelism == 0:
+            parallelism = sys.maxsize
```

Comment:
Is it a good idea if we use `math.inf` here instead? (That’d be a float though)

Reply:
I thinl "maxsize" is `good-enough` honestly  (from pragmatic point of view) ;) . I'd say it's often used technique to use 0 for "unliimited",  I personally don't feel we should add some special value only understood by us. out of necessity that would have to add parsing a conf int value from non-int , so adds complexity and has potential of bugs.

Reply:
Yupp we want this to be an int and also sys.maxsize is already what's been used historically for this:

https://github.com/aws-mwaa/upstream-to-airflow/blob/41508f23ad87bdf410978ff96e62d8057d17d9d0/airflow/executors/base_executor.py#L576-L579